pkgs/development/libraries/glib/setup-hook.sh

```diff
@@ -24,11 +24,4 @@ glibPreFixupPhase() {
     addToSearchPath GSETTINGS_SCHEMAS_PATH "${!outputLib}/share/gsettings-schemas/$name"
 }
 
-# gappsWrapperArgsHook expects GSETTINGS_SCHEMAS_PATH variable to be set by this.
-# Until we have dependency mechanism in generic builder, we need to use this ugly hack.
-if [[ " ${preFixupPhases:-} " =~ " gappsWrapperArgsHook " ]]; then
-    preFixupPhases+=" "
-    preFixupPhases="${preFixupPhases/ gappsWrapperArgsHook / glibPreFixupPhase gappsWrapperArgsHook }"
-else
-    preFixupPhases+=" glibPreFixupPhase"
-fi
+postInstallHooks+=(glibPostInstallHook)
```

Comment:
One worry is that we often need to adjust or compile the schemas in `postInstall` (see `rg share/glib-2.0/schemas -C 5`). Is it guaranteed that `glibPostInstallHook` will run after the `postInstall` passed to `mkDerivation`?

Reply:
Oh, and also need to double check the hook does not run multiple times.

Reply:
I am not sure. There were some issues where a hook would run many times ¿for every propagated copy of g-i?

Though I guess it is not the case here. Or if it is, at least it would not be a regression.